openssl-dynamic/src/main/c/ssl.c

```diff
@@ -653,20 +633,13 @@ static void ssl_set_thread_id(CRYPTO_THREADID *id)
     CRYPTO_THREADID_set_numeric(id, ssl_thread_id());
 }
 
-static apr_status_t ssl_thread_cleanup(void *data)
+static void ssl_thread_cleanup()
 {
-    UNREFERENCED(data);
     CRYPTO_set_locking_callback(NULL);
     CRYPTO_THREADID_set_callback(NULL);
     CRYPTO_set_dynlock_create_callback(NULL);
```

Comment:
it doesn't look like this method takes ownership of the memory [1]. We should take more care to free this memory, and if we set it check if there already exists something and free it.

[1] https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/cryptlib.c#L376